File: Testing/AngularTest/TestApi3/src/services/people.service.ts
import { Injectable } from '@nestjs/common';
import { PersonCreateDTO } from './dto/People/person-create.dto';
import { PersonDTO } from './dto/People/person.dto';
import { PersonUpdateDTO } from './dto/People/person-update.dto';
import { PersonRepository } from './../repository/person.repository';
import { IntentIgnore, IntentIgnoreBody } from './../intent/intent.decorators';
import { Person } from './../domain/entities/person.entity';
import { randomUUID } from 'crypto';

@Injectable()
export class PeopleService {

  //@IntentCanAdd()
  constructor(private personRepository: PersonRepository) { }

  @IntentIgnoreBody()
  async create(dto: PersonCreateDTO): Promise<string> {
    var newPerson = {
      name: dto.name,
    } as Person;

    await this.personRepository.save(newPerson);
    return newPerson.id;
  }

  @IntentIgnoreBody()
  async findById(id: string): Promise<PersonDTO> {
    return PersonDTO.fromPerson(await this.personRepository.findOneBy({ id: id }));
  }

  @IntentIgnoreBody()
  async findAll(): Promise<PersonDTO[]> {
    return await (await this.personRepository.find()).map(x => PersonDTO.fromPerson(x));
  }

  @IntentIgnoreBody()
  async update(id: string, dto: PersonUpdateDTO): Promise<void> {
    var existingPerson = await this.personRepository.findOneBy({
      id: id
    });
    existingPerson.name = dto.name;

    await this.personRepository.save(existingPerson);
  }

  @IntentIgnoreBody()
  async delete(id: string): Promise<void> {
    var existingPerson = await this.personRepository.findOneBy({
      id: id
    });
    await this.personRepository.remove(existingPerson);
  }
}

File: Tests/Testing/AngularTest/TestApi/TestApi.Application/Implementation/PeopleService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using AutoMapper;
using Intent.RoslynWeaver.Attributes;
using TestApi.Application.Interfaces;
using TestApi.Application.People;
using TestApi.Domain.Entities;
using TestApi.Domain.Repositories;

[assembly: DefaultIntentManaged(Mode.Fully)]
[assembly: IntentTemplate("Intent.Application.ServiceImplementations.ServiceImplementation", Version = "1.0")]

namespace TestApi.Application.Implementation
{
    [IntentManaged(Mode.Merge)]
    public class PeopleService : IPeopleService
    {
        private readonly IPersonRepository _personRepository;
        private readonly IMapper _mapper;

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public PeopleService(IPersonRepository personRepository, IMapper mapper)
        {
            _personRepository = personRepository;
            _mapper = mapper;
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task<Guid> Create(PersonCreateDTO dto)
        {
            var newPerson = new Person
            {
                Name = dto.Name,
            };

            _personRepository.Add(newPerson);
            await _personRepository.UnitOfWork.SaveChangesAsync();
            return newPerson.Id;
        }



        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task<PersonDTO> FindById(Guid id)
        {
            var element = await _personRepository.FindByIdAsync(id);
            return element.MapToPersonDTO(_mapper);
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task<List<PersonDTO>> FindAll()
        {
            var elements = await _personRepository.FindAllAsync();
            return elements.MapToPersonDTOList(_mapper);
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task Update(Guid id, PersonUpdateDTO dto)
        {
            var existingPerson = await _personRepository.FindByIdAsync(id);
            existingPerson.Name = dto.Name;
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task Delete(Guid id)
        {
            var existingPerson = await _personRepository.FindByIdAsync(id);
            _personRepository.Remove(existingPerson);
        }

        public void Dispose()
        {
        }
    }
}